added in chicago elec dist

bflatau opened the pull request:
Added in Chicago Elec. Dist.

PhillippOhlandt commented on lib/nerves_metal_detector/vendors/chicago_elec_dist_us.ex:
```diff
@@ -0,0 +1,100 @@
+      id: "chicagoelecdistus",
+      name: "Chicago Elec. Dist.",
+      country: :us,
+      homepage: "https://chicagodist.com/"
```
Can you remove the trailing `/`. (Damn this OCD all the time :D)

bflatau replied:
Ah, sorry!

PhillippOhlandt commented:
The parsing looks good. 

Two things:

1. Could you change the vendor module name to `ChicagoElectronicDistributorsUs` (module and file names) but keep the shorter version for the vendor `name` field?

2. Products that are in stock show a count on how many items are left, can you parse this as well? It's the same as for #30 

Product that is on stock which has an items count: https://chicagodist.com/products/12-vdc-1000ma-regulated-switching-power-adapter-ul-listed

bflatau commented:
1. I changed the names, I hope that's what you were after? 
2. I implemented a stock counter. I'm sure there's a better, more "elixir way" to do the syntax...
- I may be missing a potential error state, I'm only checking for != [] at the moment. I figure if I'm calling Floki, it will always spit out a list vs giving nil or some other value/data? 
- I'm not sure if I picked the right HTML element, but I liked that I could grab a direct value vs. doing some sort of text-to-number parsing function for the innerText since it contains both the number we want plus some text.
- Let me know what you think!

PhillippOhlandt commented on lib/nerves_metal_detector/inventory/data/product_update_items/chicago_electronic_distributors_us.ex:
```diff
@@ -0,0 +1,28 @@
+      sku: "RPI4-MODBP-2GB"
+    },
+    %ProductUpdate{
+      url: "https://chicagodist.com/collections/raspberry-pi/products/raspberry-pi-4-model-b-1gb",
```
Lets change the url to `https://chicagodist.com/products/raspberry-pi-4-model-b-1gb` to be consistent with the other urls.

bflatau replied:
Fixed! I put them in order too (to help with the OCD ha)

PhillippOhlandt commented on lib/nerves_metal_detector/vendors/chicago_electronic_distributors_us.ex:
```diff
@@ -0,0 +1,119 @@
+
+    if available_quantity != [] do
+      json_product_data =
+        Floki.find(html_tree, ".product_form")
+        |> Floki.attribute("data-variant-inventory")
+        |> Enum.at(0)
+        |> Jason.decode()
+
+      get_in(elem(json_product_data, 1) |> Enum.at(0), ["inventory_quantity"])
```
You can change the code to this to reuse the existing variable and also simplify the last line where you get the value out of the json.
```elixir
        available_quantity
        |> Enum.at(0)
        |> Jason.decode!()

      Enum.at(json_product_data, 0)["inventory_quantity"]
```

bflatau replied:
done!

PhillippOhlandt commented:
Looks good to me! Thanks!
closes #29
